presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestApproximatePercentileAggregation.java

```diff
@@ -13,432 +13,606 @@
+        testArrayAggregationLong(
                 createLongsBlock(null, 1L),
-                createLongsBlock(1L, 1L),
-                createRLEBlock(0.5, 2));
+                BlockAssertions.createRLEBlock(1, 2),
+                createRLEBlock(quantiles, 2),
+                STANDARD_ERROR,
+                quantiles,
+                1L);
 
-        assertAggregation(
-                LONG_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION,
-                2L,
+        testArrayAggregationLong(
                 createLongsBlock(null, 1L, 2L, 3L),
-                createLongsBlock(1L, 1L, 1L, 1L),
-                createRLEBlock(0.5, 4));
+                BlockAssertions.createRLEBlock(1, 4),
+                createRLEBlock(quantiles, 4),
+                STANDARD_ERROR,
+                quantiles,
+                1L, 2L, 3L);
 
-        assertAggregation(
-                LONG_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION,
-                2L,
+        testArrayAggregationLong(
                 createLongsBlock(1L, 2L, 3L),
-                createLongsBlock(1L, 1L, 1L),
-                createRLEBlock(0.5, 3));
-
-        assertAggregation(
-                LONG_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION,
-                3L,
-                createLongsBlock(1L, null, 2L, null, 2L, null, 2L, null, 3L, null, 3L, null, 3L, 4L, 5L, 6L, 7L),
-                createLongsBlock(1L, 1L, 2L, 1L, 2L, 1L, 2L, 1L, 2L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L),
-                createRLEBlock(0.5, 17));
-
-        assertAggregation(
-                LONG_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION_WITH_ACCURACY,
-                9900L,
-                createLongSequenceBlock(0, 10000),
-                createLongRepeatBlock(1, 10000),
-                createRLEBlock(0.99, 10000),
-                createRLEBlock(0.001, 10000));
+                BlockAssertions.createRLEBlock(1, 3),
+                createRLEBlock(quantiles, 3),
+                STANDARD_ERROR,
+                quantiles,
+                1L, 2L, 3L);
 
-        // weighted + array of approx_percentile
-        assertAggregation(
-                LONG_APPROXIMATE_PERCENTILE_ARRAY_WEIGHTED_AGGREGATION,
-                ImmutableList.of(1L, 2L),
-                createLongsBlock(1L, 2L, 3L),
-                createLongsBlock(4L, 2L, 1L),
-                createRLEBlock(ImmutableList.of(0.5, 0.8), 3));
+        testArrayAggregationLong(
+                createLongsBlock(1L, null, 2L, 2L, null, 2L, 2L, null, 2L, 2L, null, 3L, 3L, null, 3L, null, 3L, 4L, 5L, 6L, 7L),
+                BlockAssertions.createRLEBlock(1, 21),
+                createRLEBlock(quantiles, 21),
+                STANDARD_ERROR,
+                quantiles,
+                1L, 2L, 2L, 2L, 2L, 2L, 2L, 3L, 3L, 3L, 3L, 4L, 5L, 6L, 7L);
+
+        testArrayAggregationLong(
+                createLongSequenceBlock(-1000, 1000),
+                BlockAssertions.createRLEBlock(1, 2000),
+                createRLEBlock(quantiles, 2000),
```

Comment:
Is it necessary to add error to this commit?

Reply:
It can be a separate commit, but must be part of the PR, otherwise tests fail when we merge the two state interfaces.

Reply:
From the previous commit, these two blocks fail, the rest succeed.  Do we know where the change in behavior is coming from?  From the previous commit, it looks like a pure refactor with no change in behavior.

```java
        assertAggregation(
                FLOAT_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION_WITH_ACCURACY,
                9900.0f,
                createSequenceBlockOfReal(0, 10000),
                createLongRepeatBlock(1, 10000),
                createRLEBlock(0.99, 10000),
                createRLEBlock(0.001, 10000));
```

```java
        assertAggregation(
                LONG_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION_WITH_ACCURACY,
                9900L,
                createLongSequenceBlock(0, 10000),
                createLongRepeatBlock(1, 10000),
                createRLEBlock(0.99, 10000),
                createRLEBlock(0.001, 10000));
```

Reply:
Yes, just found it. I was initializing the underlying `QuantileDigest` with the `STANDARD_ERROR` (0.01) always, but in these cases it should've been taking a `maxError` parameter (0.001) to build the `QuantileDigest`.

Reply:
Good that we caught this.